### ISSUE TITLE
Allow for text alternative to be passed into icon in calcite-notice and calcite-alert

### ISSUE THREAD

**Issue by FelFly** [Esri/calcite-design-system#10340]
### Check existing issues

- [X] I have [checked for existing issues](https://github.com/Esri/calcite-design-system/issues) to avoid duplicates

### Summary

As color alone cannot be used to communicate status, our design team typically uses an icon on notices or alerts that belong to `kind: success, danger, warning`. However, the icon in the notice / alert does not allow us to pass in an accessible label, which was requested on a recent Hub accessibility audit. Our work-around is putting the status into the actual message as a string, but ideally, we'd be able to combine the inclusion of an icon with that status, so that the main message of the component can be consolidated.

### Actual Behavior

The icons included on the notices and alerts are set to aria-hidden="true" and we cannot change them.

![Screenshot 2024-09-18 at 9 21 06 AM](https://github.com/user-attachments/assets/56923fa5-0c05-408e-8ad6-578d327c23ff)

![Screenshot 2024-09-18 at 9 22 37 AM](https://github.com/user-attachments/assets/d79fafc8-a681-483b-83a5-c2cf60575faa)

![Screenshot 2024-09-18 at 9 27 14 AM](https://github.com/user-attachments/assets/a1666e55-9a63-4b28-a3ca-1ae8817c6669)

### Expected Behavior

We'd like to be able to pass a simple status label onto icons included in success, danger, and warning style notices and alerts, such as `success`, `warning`, `error`, `attention`. 

### Reproduction Sample

https://developers.arcgis.com/calcite-design-system/components/notice/

### Reproduction Steps

1. In either Notice or Alert, set an icon
2. Navigate component via screen-reader

### Reproduction Version

v2.12.2

### Priority impact

impact - p3 - not time sensitive

### Calcite package

- [X] @esri/calcite-components
- [ ] @esri/calcite-components-angular
- [ ] @esri/calcite-components-react
- [ ] @esri/calcite-design-tokens
- [ ] @esri/eslint-plugin-calcite-components

### Esri team

ArcGIS Hub

**Comment by driskull** [comment]
@FelFly would adding an `aria-label` to the `calcite-notice` or one of its slots work for this?

Something like:

```html
<calcite-notice kind="danger" open icon="layers-reference" aria-label="Error: the role was not updated due to insufficient permissions">
    <div slot="title">Role not updated</div>
    <div slot="message">You do not have permissions</div>
</calcite-notice>
```

or

```html
<calcite-notice kind="danger" open icon="layers-reference">
    <div slot="title" aria-label="Error: the role was not updated due to insufficient permissions">Role not updated</div>
    <div slot="message">You do not have permissions</div>
</calcite-notice>
```

I believe the icon is purely visual. Its not meant to be content.

Maybe we can add a `label` prop on the component to add an `aria-label` somewhere in the shadowDOM to make it easier.

**Comment by geospatialem** [comment]
Calcite follows [Success Criterion 1.1.1: Non-text content](https://www.w3.org/WAI/WCAG21/Understanding/non-text-content.html), where the developer can specify context through the `title` and/or `message` slots dynamically, rather than provide context to the component's supplemental or visual only `icon`. Prior to v1, Calcite added `aria-hidden` to all icons to ensure they were hidden from assistive technologies, as they are intended for their visual content.

cc @FelFly
